osu.Game.Rulesets.Catch.Tests/TestSceneHyperDashColouring.cs

```diff
@@ -130,9 +130,16 @@ private void checkHyperDashCatcherColour(ISkin skin, Color4 expectedCatcherColou
             });
 
             AddAssert("catcher colour is correct", () =>
-                expectedCatcherColour == Catcher.DEFAULT_HYPER_DASH_COLOUR
-                    ? catcherArea.MovableCatcher.Colour == Catcher.DEFAULT_CATCHER_HYPER_DASH_COLOUR
-                    : catcherArea.MovableCatcher.Colour == expectedCatcherColour);
+            {
+                var expected = expectedCatcherColour;
+
+                if (expected == Catcher.DEFAULT_HYPER_DASH_COLOUR)
+                    // The expected colour for Catcher.Colour is another colour
+                    // for the default skin, assert with that instead.
+                    expected = Catcher.DEFAULT_CATCHER_HYPER_DASH_COLOUR;
```

Comment:
this still does not make any sense to me, sorry. at this point you should be splitting out tests rather than doing this weird logic.

- what is "another colour" and why?
- why not just set it correctly to `expectedCatcherColour`?

Reply:
What do you mean by "splitting out tests", might be what I'm gonna suggest here:

I guess the best this can become would be by splitting `expectedCatcherColour` to `expectedCatcherColour, expectedTrailsColour`, and fill in accordingly.

Reply:
Basically if your test assertions have conditionals in them it means it's time to split the test into two, each of which tests each branch of the conditional assertion - so you'd test the `DEFAULT_HYPER_DASH_COLOUR` case separately and the case of a custom non-red colour separately.

That said I'd still be in favour of just getting rid of that special case.